### PR TITLE
Pika multiprocessing fix for 1.10.x

### PR DESCRIPTION
**Pull request by alwx**
**Proposed changes**:
- based on what's done in this PR: https://github.com/RasaHQ/rasa/pull/7217

What's essential here is that now we prevent the message handling process from being terminated.

**Status (please check what you already did)**:
- [ ] added some tests for the functionality
- [ ] updated the documentation
- [ ] updated the changelog (please check [changelog](https://github.com/RasaHQ/rasa/tree/master/changelog) for instructions)
- [ ] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa#code-style) for instructions)

### CODE REVIEW

File: rasa/core/brokers/pika.py [hunk 1]
```diff
@@ -498,16 +527,25 @@ def process_messages(self) -> None:
...
         starting the IOLoop to block and allow the SelectConnection to operate.
+
         This function is blocking and indefinite thus it
         should be started in a separate process.
         """
-        self._connection = self._connect()
-
-        # noinspection PyUnresolvedReferences
-        self._connection.ioloop.start()
+        # TODO: Fix logging
```
Comment (wochinge) [inline rasa/core/brokers/pika.py:545]: logging works for you right? Then let's dump this todo
```suggestion
```

File: rasa/core/brokers/pika.py [hunk 2]
```diff
@@ -602,53 +641,24 @@ def _connect(self) -> None:
...
         )
-
-        self.process = self._start_pika_process()
+        self.process = self._start_pika_connection_process()
 
     def _get_mp_context(self) -> multiprocessing.context.BaseContext:
         return multiprocessing.get_context(self.MP_CONTEXT)
 
-    def _start_pika_process(self) -> Optional[multiprocessing.Process]:
+    def _start_pika_connection_process(self) -> Optional[multiprocessing.Process]:
```
Comment (wochinge) [inline rasa/core/brokers/pika.py:652]: optional: What does the additional `connection` mean?

**Review by wochinge — COMMENTED** [review]
code looks good. Thanks for looking into this 🚀  Will give it an additional manual run and then approve 👍

**Comment by wochinge** [comment]
a bunch of tests fail btw

File: rasa/core/brokers/pika.py [hunk 3]
```diff
@@ -602,53 +641,24 @@ def _connect(self) -> None:
...
-    def _publish(self, body: Text, headers: MessageHeaders = None) -> None:
-        if not self.pika_message_processor:
-            self._connect()
-
-        if (
-            self.process and self.process.is_alive()
-        ) or self.should_keep_unpublished_messages:
-            self.process_queue.put((body, headers))
-
-    def is_ready(
```
Comment (wochinge) [inline rasa/core/brokers/pika.py:636]: we can return if the process is alive here. It's used by the `exporter` module afaik

**Review by wochinge — APPROVED** [review]
Logging works (why didn't it for 2.0? 🤔 ) and connection seems stable. Would be great if @melindaloubser1 could confirm though.

**Comment by melindaloubser1** [comment]
Yes, just tested it on microk8s, all seems good

**Review by melindaloubser1 — APPROVED** [review]

**Comment by benos** [comment]
@wochinge @alwx 

If I am reading the code correctly, this commit has silently removed the deprecated 'queue' argument:

https://github.com/RasaHQ/rasa/pull/7219/commits/253afa0f8f9cf04d78647d20b2ee10b1a3b5ce96

I imagine it was unintentional to introduce a breaking change on a patch level release, but you might want to update the changelog for 1.10.17 so that those of us still using the old argument know about it.

Also, the warning still refers to the 'queue' argument, adding to the confusion: 

UserWarning: No `queues` or `queue` argument provided. It is suggested to explicitly specify a queue as described in https://rasa.com/docs/rasa/api/event-brokers/#pika-event-broker. Using the default queue 'rasa_core_events' for now.

**Comment by wochinge** [comment]
Thanks for pointing this out @benos 
@alwx Could you please take care of it? In my opinion it would be clearer to re-add support for the old argument, and cut another micro instead of having such a major change in the changelog / code, what do you think?

**Pull request merged**